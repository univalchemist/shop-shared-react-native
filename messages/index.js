import { en_HK as shop_en_HK, zh_HK as shop_zh_HK } from '@components/shops';
import en_HK from './en-HK.json';
import en_UK from './en-UK.json';
import id_ID from './id-ID.json';
import zh_HK from './zh-HK.json';

const messages = {
  'en-HK': { ...en_HK, ...shop_en_HK },
  'en-UK': en_UK,
  'id-ID': id_ID,
  'zh-HK': { ...zh_HK, ...shop_zh_HK },
};

export default messages;
export { en_HK, en_UK, zh_HK, id_ID };
