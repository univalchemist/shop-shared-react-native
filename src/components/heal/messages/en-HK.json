{
  "available": "Available",
  "doctor": "Doctor",
  "doctor.bookAppointment": "Book an appointment",
  "doctor.scheduleAppointment": "Schedule appointment",
  "doctor.scheduleAppointment.selectFamilyMember": "Select family member",
  "doctor.scheduleAppointment.selectClinic": "Select Clinic",
  "doctor.scheduleAppointment.pickDates": "Pick Dates",
  "doctor.scheduleAppointment.maxDates": "You can add up to 3 dates",
  "doctor.scheduleAppointment.addDates": "Add Consultations dates",
  "doctor.scheduleAppointment.alert": "Are you sure you want to delete this date ?",
  "doctor.scheduleAppointment.delete": "Delete",
  "doctor.scheduleAppointment.cancel": "Cancel",
  "doctor.scheduleAppointment.pickDate": "Pick your dates",
  "doctor.scheduleAppointment.pickTime": "Pick your time",
  "doctor.scheduleAppointment.confirmDate": "Confirm Date",
  "doctor.scheduleAppointment.time.am": "AM",
  "doctor.scheduleAppointment.time.pm": "PM",
  "doctor.scheduleAppointment.confirmation": "Please confirm the information of the patient before checking-in.",
  "doctor.scheduleAppointment.next": "Next",
  "doctor.scheduleAppointment.pickDate.error": "You already have an appointment for this date Please pick another date",
  "doctor.scheduleAppointment.requestAppointment": "Request appointment",
  "doctor.scheduleAppointment.appointmentRequested": "Appointment requested",
  "doctor.scheduleAppointment.scheduleConfirmation": "You will receive a notification later to confirm the details of the appointment.",
  "doctor.scheduleAppointment.viewAppointment": "View my appointments",
  "doctor.scheduleAppointment.back": "Back to doctor details",
  "doctorSearch.placeholder": "Search doctor or location",
  "doctor.found": "{total} doctors found",
  "doctor.workIn": "Works in {number} clinic(s)",
  "unavailable": "Unavailable",
  "speciality.name.generalpractitio": "General Practitioner",
  "speciality.name.generalsurgery": "General Surgery",
  "speciality.name.gastroenterology": "Gastroenterology & Hepatology",
  "speciality.name.administrativeme": "Administrative Medicine",
  "speciality.name.anaesthesiology": "Anaesthesiology",
  "speciality.name.cardiology": "Cardiology",
  "speciality.name.cardiothoracicsu": "Cardio-thoracic Surgery",
  "speciality.name.chiropractor": "Chiropractor",
  "speciality.name.clinicalmicrobio": "Clinical Microbiology & Infection",
  "speciality.name.clinicaloncology": "Clinical Oncology",
  "speciality.name.communitymedicin": "Community Medicine",
  "speciality.name.criticalcaremedi": "Critical Care Medicine",
  "speciality.name.dentist": "Dentist",
  "speciality.name.dermatologyvener": "Dermatology & Venereology",
  "speciality.name.developmentalbeh": "Developmental-Behavioural Paediatrics",
  "speciality.name.emergencymedicin": "Emergency Medicine",
  "speciality.name.counsellingpsych": "Counseling Psychology",
  "speciality.name.geneticsandgenom": "Genetics and Genomics (Pediatric)",
  "speciality.name.paediatrics": "Paediatrics",
  "speciality.name.urogynaecology": "Urology",
  "speciality.name.orthopaedicstrau": "Orthopaedics & Traumatology",
  "specialist": "Specialist",
  "specialist.placeholder": "Search for a doctor",
  "findSpecialist": "Find a specialist",
  "clinic": "Clinic",
  "clinic.pluralFound": "{number} clinics found",
  "clinic.singularFound": "{number} clinic found",
  "clinic.noClinics": "No clinics available",
  "clinic.noClinicsDescription": "There are no clinics matching your search criteria.",
  "clinic.pluralDoctor": "{number} Doctors",
  "clinic.singularDoctor": "{number} Doctor",
  "heal.clinic.mdistance": "{number}m away",
  "heal.clinic.kmdistance": "{number}km away",
  "clinic.doctors": "Doctors",
  "clinic.clinicStatus": "Clinic Status",
  "clinic.queueButton": "Get a queue",
  "clinic.open": "Open",
  "clinic.closed": "Closed",
  "clinic.estimatedTime": "estimated consulation time",
  "clinic.openingHours.Sun": "Sunday",
  "clinic.openingHours.Mon": "Monday",
  "clinic.openingHours.Tue": "Tuesday",
  "clinic.openingHours.Wed": "Wednesday",
  "clinic.openingHours.Thu": "Thursday",
  "clinic.openingHours.Fri": "Friday",
  "clinic.openingHours.Sat": "Saturday",
  "clinic.openingHours.Public Holiday": "Public Holiday",
  "heal.NearMe": "Near me",
  "heal.UnifySearch.NotFound": "There are no clinics or doctors matching your search criteria.",
  "heal.UnifySearch.placeholder": "Search clinic or location",
  "heal.UnifySearch.DoctorName": "Doctor name",
  "heal.UnifySearch.ClinicName": "Clinic name",
  "heal.UnifySearch.Location": "Location",
  "heal.UnifySearch.Speciality": "Speciality",
  "heal.UnifySearch.Recent": "Recent",
  "heal.checkin.SelectMember": "Select family member",
  "heal.checkin.NoticeConfirmation": "Please confirm the information of the patient before checking-in.",
  "heal.next": "Next",
  "heal.checkin.title": "Check-in",
  "thankYouScreen.headText": "Thank you for booking with us!",
  "thankYouScreen.timeText": "Estimation consultation time is at {time} on {date}",
  "thankYouScreen.buttonText": "View my appointments",
  "appointmentDetailScreen.headText": "The Clinic Group @Fusionopolis",
  "appointmentDetailScreen.rejectBtnText": "Reject",
  "appointmentDetailScreen.acceptBtnText": "Accept",
  "appointmentDetailScreen.upComingBtnText": "Cancel appointment",
  "appointmentConfirmation.pendingConfirmationTittle": "Appointment is Confirmed",
  "appointmentConfirmation.pendingConfirmationMessage": "Your appointment schedule has been confirmed!",
  "appointmentConfirmation.rejectTittle": "Confirmation",
  "appointmentConfirmation.rejectMessage": "Are you sure you want to cancel this appointment schedule?",
  "appointmentConfirmation.confirm": "Confirm",
  "appointmentConfirmation.cancel": "Cancel",
  "appointmentConfirmation.Ok": "Ok",
  "healTabNavigator.findDoctor": "Find a doctor",
  "healTabNavigator.appointment": "Appointments",
  "healTabNavigator.medication": "Medications",
  "heal.CheckInForm.fullname.label": "Fullname (as in NRIC / HK ID)*",
  "heal.CheckInForm.chineseFullname.label": "Chinese fullname",
  "heal.CheckInForm.idNumber.label": "Identification number*",
  "heal.CheckInForm.phoneNumber.label": "Phone number",
  "heal.CheckInForm.email.label": "Email address",
  "heal.CheckInForm.dateOfBirth.label": "Date of birth*",
  "heal.CheckInForm.address.label": "Address",
  "heal.CheckInForm.contactPerson.label": "Contact person",
  "heal.CheckInForm.relationship.label": "Relationship",
  "heal.CheckInForm.contactNumber.label": "Contact number",
  "heal.CheckInForm.description": "A remote ticketing deposit of $1 will be charged. It will be refunded after your doctorâ€™s visit. If you cancel withn 60min of your estimated consultation time or no-show, the deposit will not be refunded.",
  "heal.myself": "Myself",
  "heal.CheckInForm.required": "This field is required",
  "heal.gender.male": "Male",
  "heal.gender.female": "Female",
  "heal.gender": "Gender",
  "heal.walkInCheckIn.back": "Back to homepage",
  "heal.walkInCheckIn.estimatedConsultation": "Estimated consulation time",
  "heal.walkInCheckIn.title": "You're checked-in",
  "heal.walkInSelectDoctor.selectTitle": "Select your doctor",
  "heal.walkInSelectDoctor.confirmButton": "Confirm",
  "appointment.doctorPrefix": "Dr.",
  "appointmentListingScreen.noData": "No appointments yet.",
  "heal.CheckInForm.confirmAndNext": "Confirm & Next",
  "heal.backToHomepage": "Back to homepage",
  "heal.confirmation.checkedIn": "You're checked-in",
  "heal.CheckIn.estConsultationTime": "Estimated consultation time",
  "heal.CheckIn.notice": "Please check the details of your appointment today.",
  "heal.CheckIn.seeAnotherDoctors": "See another doctor",
  "heal.confirmNextScreen.heading": "Please check the detail of your appointment today.",
  "heal.confirmNextScreen.subHeading": "Appointment Schedule",
  "heal.confirmNextScreen.anotherDoctor": "See another doctor",
  "heal.confirmNextScreen.btnText": "Confirm & Next"
}
