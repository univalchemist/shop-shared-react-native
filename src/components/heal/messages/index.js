import en_HK from './en-HK.json';
import zh_HK from './zh-HK.json';

const messages = {
  'en-HK': en_HK,
  'zh-HK': zh_HK,
};

export default messages;
export { en_HK, zh_HK };
