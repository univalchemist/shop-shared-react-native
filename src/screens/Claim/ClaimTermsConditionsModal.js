import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import {
  ScrollView,
  Container,
  Box,
  TrackedButton,
  Loader,
  PlainText,
  Footer,
} from '@wrappers/components';
import { submitClaim, uploadDocumentsForReference } from '@store/claim/actions';
import { compose } from 'redux';
import { reduxForm } from 'redux-form';
import { CLAIMS_LIST, CLAIM_SUCCESS_MODAL, CLAIM_ERROR_MODAL } from '@routes';
import { injectIntl } from 'react-intl';
import { categories } from '@store/analytics/trackingActions';

const TcContainer = styled(Container)`
  ${({ theme }) => `
   background-color: ${theme.backgroundColor.default}
   padding-bottom: 0
   flex: 1
  `}
`;

const ClaimTermsConditionsModal = ({
  handleSubmit,
  submitClaim,
  uploadDocumentsForReference,
  submitting,
  intl,
  navigation,
}) => {
  const handleClaimSubmit = async (values, _, { reset }) => {
    let response;
    let receiptFilesIds;
    let referralFilesIds;
    let settlementAdviceIds;
    let prescriptionIds;

    try {
      const { documents } = values;
      response = await uploadDocumentsForReference('Receipt', documents);
      receiptFilesIds = response.action.payload;
    } catch (error) {
      return navigation.navigate(CLAIM_ERROR_MODAL);
    }

    try {
      const { referrals } = values;
      if (referrals && referrals.length > 0) {
        response = await uploadDocumentsForReference('Referral', referrals);
        referralFilesIds = response.action.payload;
      }
    } catch (error) {
      return navigation.navigate(CLAIM_ERROR_MODAL);
    }

    try {
      const { settlementAdvices } = values;
      if (settlementAdvices && settlementAdvices.length > 0) {
        response = await uploadDocumentsForReference(
          'SettlementAdvices',
          settlementAdvices,
        );
        settlementAdviceIds = response.action.payload;
      }
    } catch (error) {
      return navigation.navigate(CLAIM_ERROR_MODAL);
    }

    try {
      const { prescriptions } = values;
      if (prescriptions && prescriptions.length > 0) {
        response = await uploadDocumentsForReference(
          'Prescriptions',
          prescriptions,
        );
        prescriptionIds = response.action.payload;
      }
    } catch (error) {
      return navigation.navigate(CLAIM_ERROR_MODAL);
    }

    try {
      const result = await submitClaim({
        ...values,
        receiptFilesIds,
        referralFilesIds,
        settlementAdviceIds,
        prescriptionIds,
      });
      const { claimId } = result.action.payload;
      const state = navigation.dangerouslyGetState();
      const firstRoute = state?.routes?.[0];

      await navigation.reset({
        routes: [
          { name: firstRoute?.name || CLAIMS_LIST },
          { name: CLAIM_SUCCESS_MODAL, params: { claimId } },
        ],
      });

      reset();
    } catch (error) {
      return navigation.navigate(CLAIM_ERROR_MODAL);
    }
  };

  if (submitting) {
    return (
      <Loader
        loadingText={intl.formatMessage({ id: 'claim.claimSubmitLoadingText' })}
      />
    );
  }

  return (
    <Box flex={1}>
      <ScrollView>
        <TcContainer>
          <Box marginBottom={32}>
            <PlainText>
              {intl.formatMessage({ id: 'claim.termsConditionsText' })}
            </PlainText>
          </Box>
        </TcContainer>
      </ScrollView>
      <Footer>
        <TrackedButton
          primary
          title={intl.formatMessage({ id: 'claim.acceptClaimTermsConditions' })}
          onPress={handleSubmit(handleClaimSubmit)}
          actionParams={{
            category: categories.CLAIMS,
            action: 'Accept claims t&cs',
          }}
        />
      </Footer>
    </Box>
  );
};

ClaimTermsConditionsModal.propTypes = {
  values: PropTypes.shape({}),
  submitClaim: PropTypes.func,
  handleSubmit: PropTypes.func,
  submitting: PropTypes.bool,
  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
  uploadDocumentsForReference: PropTypes.func,
};

const enhance = compose(
  connect(null, { submitClaim, uploadDocumentsForReference }),
  reduxForm({
    form: 'claimDetailsForm',
    destroyOnUnmount: false,
  }),
  injectIntl,
);

export default enhance(ClaimTermsConditionsModal);
