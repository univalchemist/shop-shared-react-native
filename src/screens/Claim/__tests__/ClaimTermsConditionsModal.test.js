import React from 'react';
import { renderForTest } from '@testUtils';
import {
  fireEvent,
  flushMicrotasksQueue,
  act,
} from 'react-native-testing-library';
import ClaimTermsConditionsModal from '../ClaimTermsConditionsModal';
import { CLAIMS_LIST, CLAIM_SUCCESS_MODAL, CLAIM_ERROR_MODAL } from '@routes';
import messages from '@messages/en-HK.json';
import originalNavigation from '@testUtils/__mocks__/navigation';

const navigation = {
  ...originalNavigation,
  dangerouslyGetState: jest.fn().mockImplementation(() => ({
    routes: [
      {
        routeName: CLAIMS_LIST,
      },
    ],
  })),
};

const api = {
  uploadDocumentReference: jest.fn().mockImplementation(() => ({
    data: {
      documentId: 'document-id',
    },
  })),
  submitClaim: jest.fn().mockImplementation(() => ({
    data: {
      claimId: 'claim-id',
    },
  })),
};

const initialState = {
  claimType: {
    categories: {
      all: [1],
      byId: {
        1: {
          claimCategory: 'Outpatient',
          claimTypeIds: [1],
          code: 'outpatient',
          displayOrder: 1,
          id: 1,
          isInsuranceClaim: true,
        },
      },
    },
    reasons: {
      byId: {
        1: { claimReason: 'Abdominal Colic', code: 'COLIC', id: 1 },
      },
    },
    types: {
      byId: {
        1: {
          claimCategoryId: 1,
          claimReasons: [1],
          claimType: 'General practitioner',
          code: 'MO-GP',
          id: 1,
          isInsuranceClaim: true,
          maxAmountPerClaim: 800,
          referralRequired: false,
        },
      },
    },
  },
};

describe('ClaimTermsConditionsModal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it.skip('should render correctly', () => {
    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState,
        api,
      },
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it.skip('should render loading screen', () => {
    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState: {
          form: { claimDetailsForm: { submitting: true } },
          ...initialState,
        },
        api,
      },
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should navigate to CLAIM_REVIEW on uploadDocumentReference type:Receipt error', async () => {
    api.uploadDocumentReference.mockImplementationOnce(() => {
      const error = {
        response: { data: { message: 'foo' } },
      };
      throw error;
    });

    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState: {
          form: {
            claimDetailsForm: {
              values: {
                documents: [{ uri: 'some-receipt' }],
                referrals: [],
                claimTypeId: 1,
                claimReason: 1,
              },
            },
          },
          ...initialState,
        },
        api,
      },
    );

    const button = component.getByProps({
      title: messages['claim.acceptClaimTermsConditions'],
    });

    act(() => {
      fireEvent.press(button);
    });
    await flushMicrotasksQueue();

    expect(navigation.navigate).toHaveBeenCalledWith(CLAIM_ERROR_MODAL);
  });

  it('should navigate to CLAIM_REVIEW on uploadDocumentReference type:Referral error', async () => {
    api.uploadDocumentReference
      .mockImplementationOnce(() => ({
        // Mock for type Receipt
        data: {
          documentId: 'document-id',
        },
      }))
      .mockImplementationOnce(() => {
        // Mock for type Referral
        const error = {
          response: { data: { message: 'foo' } },
        };
        throw error;
      });

    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState: {
          form: {
            claimDetailsForm: {
              values: {
                claimTypeId: 1,
                claimReason: 1,
                documents: [{ uri: 'some receipt' }],
                referrals: [
                  {
                    uri: 'some referral',
                  },
                ],
              },
            },
          },
          ...initialState,
        },
        api,
      },
    );

    const button = component.getByProps({
      title: messages['claim.acceptClaimTermsConditions'],
    });

    act(() => {
      fireEvent.press(button);
    });
    await flushMicrotasksQueue();

    expect(navigation.navigate).toHaveBeenCalledWith(CLAIM_ERROR_MODAL);
  });

  it('should navigate to CLAIM_REVIEW on submitClaim error', async () => {
    api.submitClaim.mockImplementationOnce(values => {
      const error = {
        response: { data: { message: 'foo' } },
      };
      throw error;
    });

    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState: {
          form: {
            claimDetailsForm: {
              values: {
                claimTypeId: 1,
                claimReason: 1,
                claimantId: 'Born to fail',
                documents: [],
                referrals: [],
              },
            },
          },
          ...initialState,
        },
        api,
      },
    );
    const button = component.getByProps({
      title: messages['claim.acceptClaimTermsConditions'],
    });

    act(() => {
      fireEvent.press(button);
    });
    await flushMicrotasksQueue();

    expect(api.submitClaim).toHaveBeenCalled();
    expect(navigation.navigate).toHaveBeenCalledWith(CLAIM_ERROR_MODAL);
  });

  it('should navigate to CLAIM_SUCCESS_MODAL on success', async () => {
    const component = renderForTest(
      <ClaimTermsConditionsModal navigation={navigation} />,
      {
        initialState: {
          ...initialState,
          claim: { isSubmitting: false },
          form: {
            claimDetailsForm: {
              values: {
                claimTypeId: 1,
                claimantId: '3',
                claimReason: 1,
                documents: [{ uri: 'receipt1' }],
                referrals: [{ uri: 'referral1' }],
              },
            },
          },
        },
        api,
      },
    );
    const button = component.getByProps({
      title: messages['claim.acceptClaimTermsConditions'],
    });

    act(() => {
      fireEvent.press(button);
    });
    await flushMicrotasksQueue();
    expect(navigation.reset).toHaveBeenCalledWith({
      routes: [
        { name: 'ClaimsList' },
        { name: 'ClaimSuccessModal', params: { claimId: 'claim-id' } },
      ],
    });
    expect(api.uploadDocumentReference).toHaveBeenCalledTimes(2);
    expect(api.uploadDocumentReference).toHaveBeenCalledWith(
      undefined,
      undefined,
      'Receipt',
      {
        uri: 'receipt1',
      },
    );
    expect(api.uploadDocumentReference).toHaveBeenCalledWith(
      undefined,
      undefined,
      'Referral',
      {
        uri: 'referral1',
      },
    );
  });
});
