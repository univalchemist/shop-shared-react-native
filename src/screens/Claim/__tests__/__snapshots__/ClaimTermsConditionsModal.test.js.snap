// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimTermsConditionsModal should render correctly 1`] = `
<View
  flex={1}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    flex={1}
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#f9f9f9",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      innerRef={[Function]}
      keyboardDismissMode="on-drag"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      onScrollBeginDrag={null}
      resetKeyboardSpace={[Function]}
      scrollEventThrottle={16}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f9f9f9",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 0,
                "paddingLeft": 32,
                "paddingRight": 32,
                "paddingTop": 32,
              },
            ]
          }
        >
          <View
            marginBottom={32}
            style={
              Array [
                Object {
                  "marginBottom": 32,
                },
              ]
            }
          >
            <Text
              color="gray.1"
              fontFamily="default"
              fontSize={16}
              letterSpacing={0.3}
              lineHeight={22}
              style={
                Array [
                  Object {
                    "color": "#646464",
                    "fontFamily": "Open Sans",
                    "fontSize": 16,
                    "letterSpacing": 0.3,
                    "lineHeight": 22,
                  },
                ]
              }
            >
              PERSONAL INFORMATION COLLECTION STATEMENT

AXA General Insurance Hong Kong Limited (referred to hereinafter as the “Company”) recognises its responsibilities in relation to the collection, holding, processing, use and/or transfer of personal data under the Personal Data (Privacy) Ordinance (Cap. 486) (“PDPO”).  Personal data will be collected only for lawful and relevant purposes and all practicable steps will be taken to ensure that personal data held by the Company is accurate.  The Company will take all practicable steps to ensure security of the personal data and to avoid unauthorised or accidental access, erasure or other use.

Please note that if you do not provide us with your personal data, we may not be able to provide the information, products or services you need or process your request.

PURPOSE:  From time to time it is necessary for the Company to collect your personal data which may be used, stored, processed, transferred, disclosed or shared by us for purposes (“Purposes”), including:

1. processing and evaluating any applications or requests made by you for products/services offered by the Company and other companies of the AXA Group (“our affiliates”); 

2. providing subsequent services to you, including but not limited to administering the policies issued; 

3. any purposes in connection with any claims made by or against or otherwise involving you in respect of any products/services provided by the Company and/or our affiliates, including investigation of claims; 

4. evaluating your financial needs; 

5. designing products/services for customers; 

6. conducting market research for statistical or other purposes; 

7. matching any data held which relates to you from time to time for any of the purposes listed herein; 

8. making disclosure as required by any applicable law, rules, regulations, codes of practice or guidelines or to assist in law enforcement purposes, investigations by police or other government or regulatory authorities in Hong Kong or elsewhere; 

9. conducting identity and/or credit checks and/or debt collection; 

10. complying with the laws of any applicable jurisdiction; 

11. carrying out other services in connection with the operation of the Company’s business; and 

12. other purposes directly relating to any of the above. 

TRANSFER OF PERSONAL DATA:  Personal data will be kept confidential but, subject to the provisions of any applicable law, may be provided to:

1. any of our affiliates, any person associated with the Company, any reinsurance company, claims investigation company, your broker, industry association or federation, fund management company or financial institution in Hong Kong or elsewhere and in this regard you consent to the transfer of your data outside of Hong Kong; 

2. *The Hong Kong and Shanghai Banking Corporation Limited (“HSBC”) for any of the Purposes and for the following additional bank related purposes: ensuring ongoing credit worthiness of customers, creating and maintaining credit and risk related models, providing the personal data to credit reference agencies for the purposes of conducting credit checks and other directly related purposes, determining the amount of indebtedness owed to or by customers and collection of amounts outstanding from customers and those providing security for customers’ obligations; 

3. any person (including private investigators) in connection with any claims made by or against or otherwise involving you in respect of any products/services provided by the Company and/or our affiliates; 

4. any agent, contractor or third party who provides administrative, technology or other services to the Company and/or our affiliates in Hong Kong or elsewhere and who has a duty of confidentiality to the same; 

5. credit reference agencies or, in the event of default, debt collection agencies; 

6. any actual or proposed assignee, transferee, participant or sub-participant of our rights or business; and 

7. any government department or other appropriate governmental or regulatory authority in Hong Kong or elsewhere. 

Transfer of your personal data will only be made for one or more of the Purposes specified above. 

ACCESS AND CORRECTION OF PERSONAL DATA: Under the PDPO, you have the right to ascertain whether the Company holds your personal data, to obtain a copy of the data, and to correct any data that is inaccurate. You may also request the Company to inform you of the type of personal data held by it. 

Requests for access and correction or for information regarding policies and practices and kinds of data held by the Company should be addressed in writing to: 

Data Privacy Officer - Employee Benefits

AXA China Region Insurance Company Limited / AXA General Insurance Hong Kong Limited, 

Unit 2201 – 2206, 22/F Manhattan Place, 23 Wang Tai Road, Kowloon Bay, Hong Kong

A reasonable fee may be charged to offset the Company’s administrative and actual costs incurred in complying with your data access requests.

Declaration and authorisation

I/WE ACKNOWLEDGE AND CONFIRM that I/we have read and understood the Personal Information Collection Statement (“PICS”).  I/We confirm that I/we have been advised to read carefully the PICS, and I/we have read it carefully its effect and impact in respect of my/our personal data collected or held by the Company (whether contained in this application or otherwise).  Based on the foregoing, I/we hereby give my/our acknowledgement and agree to the use and transfer of my/our personal data by AXA General Insurance Hong Kong Limited in accordance with the PICS.

I/WE HEREBY DECLARE AND AGREE on behalf of myself or other person referred to in this e-submission that all statements and answers to all questions are to the best of my/our knowledge and belief complete and true.

I/WE HEREBY AUTHORISE that (1) any employer, registered medical practitioner, hospital, clinic, insurance company, bank, government institution, or other organization, institution or person, that has any records or knowledge of me/us to disclose such information to the Company as the Company may request; (2) the Company or any of its appointed medical examiners or laboratories to perform the necessary medical assessment and tests to evaluate the health status of myself/ourselves in relation to this application and any claim arising therefrom.  This authorization shall bind the successors an assignees of the Relevant Persons and remains valid notwithstanding death or incapacity.

I/WE HEREBY AGREE that the company or any of its appointed representatives reserves the right to request for the original receipt within 6 months as of the completion date of the claims result.  I/We will submit the original receipt within 10 working days upon receipt of the company or authorized representative’s request.  If the company to the authorized representative does to receive the original receipt as requested within the required timeline, then the Company reserves the right to request for a refund of the payment or reverse such payment against future claims.

In the event of inconsistency between the English version and the Chinese version, the English version shall prevail.
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Array [
        Object {
          "borderTopColor": "#e3e3e3",
          "borderTopWidth": 1,
          "flexDirection": undefined,
          "paddingBottom": 16,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessibilityLabel="Accept Terms & Conditions"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#50ABAD",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Open Sans",
                "fontSize": 16,
                "lineHeight": 20,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Accept Terms & Conditions
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ClaimTermsConditionsModal should render loading screen 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9a9a9a",
            "fontSize": 128,
            "letterSpacing": -20,
            "marginTop": -20,
            "opacity": 0.4,
          }
        }
      >
         
        .
      </Text>
      <Text
        style={
          Object {
            "color": "#9a9a9a",
            "fontSize": 128,
            "letterSpacing": -20,
            "marginTop": -20,
            "opacity": 0.4,
          }
        }
      >
         
        .
      </Text>
      <Text
        style={
          Object {
            "color": "#9a9a9a",
            "fontSize": 128,
            "letterSpacing": -20,
            "marginTop": -20,
            "opacity": 0.4,
          }
        }
      >
         
        .
      </Text>
    </View>
    <Text
      color="gray.1"
      fontFamily="default"
      fontSize={16}
      lineHeight={22}
      style={
        Array [
          Object {
            "color": "#646464",
            "fontFamily": "Open Sans",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#646464",
            "fontSize": 16,
          },
        ]
      }
    >
      Submitting your claim...
    </Text>
  </View>
</Modal>
`;
